src/libraries/System.Private.CoreLib/src/System/Collections/Generic/ArraySortHelper.cs

```diff
@@ -991,4 +1011,87 @@ private static void InsertionSort(Span<TKey> keys, Span<TValue> values)
     }
 
     #endregion
+
+    /// <summary>Helper methods for use in array/span sorting routines.</summary>
+    internal static class SortUtils
+    {
+        // - These methods exist for use in sorting, where the additional operations present in
+        //   the CompareTo methods that would otherwise be used on these primitives add non-trivial overhead,
+        //   in particular for floating point where the CompareTo methods need to factor in NaNs.
+        // - The floating-point comparisons here assume no NaNs, which is valid only because the sorting routines
+        //   themselves special-case NaN with a pre-pass that ensures none are present in the values being sorted
+        //   by moving them all to the front first and then sorting the rest.
+        // - An alternative to address code quality for the floating-point values is to instead use a new wrapper
+        //   type that contains the float/double and implements CompareTo to assume no-NaNs.  That, however,
+        //   isn't quite as good, as it means the comparison operation needs to implement the CompareTo semantics
+        //   rather than just the LessThan or GreaterThan semantics needed at a given call site.
+        // - The `? true : false` is to work-around poor codegen until the above is fixed:
+        //   https://github.com/dotnet/runtime/issues/37904#issuecomment-644180265.
```

Comment:
@stephentoub I didn't know this workaround. I tried the below trick but got hit by the mentioned issue. 😅 Nice.